proxy/receiver_proxy.go

```diff
@@ -99,7 +102,7 @@ func NewReceiverProxy(config ReceiverProxyConfig) (*ReceiverProxy, error) {
 		return nil, err
 	}
 
-	cert, key, err := utils_tls.GenerateTLS(config.CertValidDuration, config.CertHosts)
+	cert, key, err := utils_tls.GetOrGenerateTLS(config.CertPath, config.CertKeyPath, config.CertValidDuration, config.CertHosts)
```

Comment:
Did you consider having two separate functions for this? I think it's fine either way, but maybe better if separate
Edit: I see the old GenerateTLS is still there, so if you'd want to allow epheremral certs you'd want to check if cert paths are empty and in that case call it instead